File: madpractical4/app/src/main/java/sg/edu/np/mad/madpractical4/ListActivity.java
package sg.edu.np.mad.madpractical4;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.Random;

public class ListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_list);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("Profile");
        builder.setMessage("MADness");
        builder.setNegativeButton("CLOSE", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        builder.setPositiveButton("VIEW", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Intent openActivity = new Intent(ListActivity.this, MainActivity.class);
                int random = new Random().nextInt(999999);
                openActivity.putExtra("ranVal",random);
                startActivity(openActivity);
            }
        });
        ImageView imgClick = findViewById(R.id.ImageClick);
        imgClick.setOnClickListener(v ->{
            AlertDialog alert = builder.create();
            alert.show();
        });

    }
}

File: app/src/main/java/sg/edu/np/mad/madpractical4/ListActivity.kt
package sg.edu.np.mad.madpractical4

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.util.Random

class ListActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_list)
        ViewCompat.setOnApplyWindowInsetsListener(
            findViewById(R.id.main)
        ) { v: View, insets: WindowInsetsCompat ->
            val systemBars =
                insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Profile")
        builder.setMessage("MADness")
        builder.setNegativeButton(
            "CLOSE"
        ) { dialog, which -> dialog.dismiss() }
        builder.setPositiveButton(
            "VIEW"
        ) { dialog, which ->
            val openActivity = Intent(
                this@ListActivity,
                MainActivity::class.java
            )
            val random = Random().nextInt(999999)
            openActivity.putExtra("ranVal", random)
            startActivity(openActivity)
        }
        val imgClick = findViewById<ImageView>(R.id.ImageClick)
        imgClick.setOnClickListener { v: View? ->
            val alert = builder.create()
            alert.show()
        }
    }
}